### PR TITLE
Bugfix/kill metadata service

### PR DESCRIPTION
**Pull request by katherinebernstein**

### CODE REVIEW

File: conductor-client/src/main/java/com/openlattice/postgres/PostgresColumn.java [hunk 1]
```diff
@@ -336,44 +336,47 @@
     public static final String                   LINKED_FIELD                      = "linked";
     public static final PostgresColumnDefinition LINKED                            =
             new PostgresColumnDefinition( LINKED_FIELD, BOOLEAN ).notNull();
-    public static final String                   LINKING_FIELD                     = "linking";
-    public static final PostgresColumnDefinition LINKING                           =
+    public static final String                   LINKING_FIELD        = "linking";
```
[the remark this answers is not in the document]
Comment (katherinebernstein) [reply conductor-client/src/main/java/com/openlattice/postgres/PostgresColumn.java:339]: i'll autoformat

File: api/src/main/kotlin/com/openlattice/datasets/DatasetMetadataApi.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    /**
+     * Gets dataset columns as a map using their aclKeys
+     *
+     * @param datasetColumnAclKeys The aclKeys of the dataset columns to load
+     *
+     * @return A map from dataset column [AclKey] to [DatasetColumn]
+     */
+    @POST(BASE + COLUMN_PATH)
+    fun getDatasetColumns(@Body datasetColumnAclKeys: Set<AclKey>): Map<AclKey, DatasetColumn>
```
[the remark this answers is not in the document]
Comment (katherinebernstein) [reply api/src/main/kotlin/com/openlattice/datasets/DatasetMetadataApi.kt:67]: I like it! adding

File: conductor-client/src/main/kotlin/com/openlattice/datastore/services/EntitySetManager.kt [hunk 3]
```diff
@@ -85,7 +85,9 @@ interface EntitySetManager {
 
     fun updateEntitySetPropertyMetadata(entitySetId: UUID, propertyTypeId: UUID, update: MetadataUpdate)
 
-    fun updateEntitySetMetadata(entitySetId: UUID, update: MetadataUpdate)
+    fun updateEntitySet(entitySetId: UUID, update: MetadataUpdate)
+
+    fun updateEntitySetMetadata(entitySetId: UUID, update: MetadataUpdate): EntitySet
```
[the remark this answers is not in the document]
Comment (katherinebernstein) [reply conductor-client/src/main/kotlin/com/openlattice/datastore/services/EntitySetManager.kt:90]: yes, they do different things and we need the granularity.

[the remark this answers is not in the document]
Comment (katherinebernstein) [reply conductor-client/src/main/kotlin/com/openlattice/datastore/services/EntitySetManager.kt:90]: yes, they do different things and we need the granularity.

File: conductor-client/src/main/kotlin/com/openlattice/datastore/services/EntitySetService.kt [hunk 4]
```diff
@@ -371,10 +370,10 @@ class EntitySetService(
                 .map { propertyTypeId -> AclKey(entitySet.id, propertyTypeId) }
                 .forEach { aclKey ->
                     authorizations.deletePermissions(aclKey)
-                    entitySetPropertyMetadata.delete(EntitySetPropertyKey(aclKey[0], aclKey[1]))
```
[the remark this answers is not in the document]
Comment (katherinebernstein) [reply conductor-client/src/main/kotlin/com/openlattice/datastore/services/EntitySetService.kt:374]: it's two lines below :)

File: conductor-client/src/main/kotlin/com/openlattice/datastore/services/EntitySetService.kt [hunk 5]
```diff
@@ -611,44 +620,47 @@ class EntitySetService(
...
-                .mapValues { it.value.associateBy({ e -> e.key.propertyTypeId }, { e -> e.value }) }
+                .groupBy { it.key[0] }
+                .mapValues {
+                    it.value.associateBy({ e -> e.key[1] }, { e ->
+                        EntitySetPropertyMetadata(
+                                e.value.title,
+                                e.value.description,
+                                e.value.flags.mapTo(linkedSetOf()) { f -> f },
+                                true
+                        )
```
[the remark this answers is not in the document]
Comment (katherinebernstein) [reply conductor-client/src/main/kotlin/com/openlattice/datastore/services/EntitySetService.kt:653]: yeah og gallery is the only one, as far as I know

**Review by makosblade — DISMISSED** [review]

**Review by makosblade — APPROVED** [review]

**Review by UnsungHero97 — APPROVED** [review]

**Pull request merged**